bodhi/server/models.py

```diff
@@ -549,6 +550,45 @@ def get_pkg_pushers(self, branch, settings):
 
         return (committers, watchers), (committergroups, watchergroups)
 
+    def get_pkg_committers_from_pagure(self, settings):
+        """
+        Pull users who can commit and are watching a package in Pagure.
+
+        Return two two-tuples of lists:
+        * The first tuple is for usernames.  The second tuple is for groups.
+        * The first list of the tuple is for committers. The second is for
+          watchers.
+        """
+        namespace = 'rpms'
+        package_pagure_url = '{0}/api/0/{1}/{2}'.format(
+            settings['pagure_url'].rstrip('/'), namespace, self.name)
+        package_json = requests.get(package_pagure_url)
+        if package_json.status_code != 200:
+            raise BodhiException(
+                'The specified package was not found in Pagure')
+        package_json = package_json.json()
+        committers = set()
+        for access_type in ['owner', 'admin', 'commit']:
+            committers = committers | set(
+                package_json['access_users'][access_type])
+        for access_type in ['admin', 'commit']:
+            for group_name in package_json['access_groups'][access_type]:
+                group_pagure_url = '{0}/api/0/group/{1}'.format(
+                    settings['pagure_url'].rstrip('/'), group_name)
+                group_json = requests.get(group_pagure_url).json()
+                committers = committers | set(group_json['members'])
+
+        watchers = set()
+        package_watchers_pagure_url = '{0}/watchers'.format(package_pagure_url)
+        watchers_json = requests.get(package_watchers_pagure_url).json()
+        for watcher, watcher_levels in watchers_json['watchers'].items():
+            # TODO: Should this include issue watchers, commit watchers, or
+            # both?
+            if 'issues' in watcher_levels:
+                watchers.add(watcher)
+
+        return (list(committers), list(watchers)), ([], [])
```

Comment:
Pagure only supports local groups and FAS groups which is why this is empty. Let me add a comment here to make this clear.